### ISSUE TITLE
csharp generator v3 does not generate .Net clients

### ISSUE THREAD

**Issue by borota** [Azure/autorest.csharp#1044]
**I was prompted to upgrade AutoRest to latest. Generated code now targets Azure only**
```
autorest readme.md
```
contents of `readme.md`
```
input-file: https://internal.site/api/swagger/docs/v0
output-folder: ..\..\OpenApi\Generated
add-credentials: true
use-internal-constructors: true
sync-methods: none
csharp:
  namespace: OpenApi
```

**Expected behavior**
Auto-generated code targeting .NET

**Additional context**
I didn't see anything anywhere about support for generating clients targeting .NET having been dropped. The only pointer I found was about using `version=previous`, if having problems with current version. What do I need to add to make sure vanilla .NET clients are being generated, using V3?

**Comment by aanufriyev** [comment]
Any updates on this? We are actively using AutoRest to generate internal clients and if it is confirmed that non-Azure template will be dropped we will have to plan switch now, instead of using --v2 flag for some time.

**Comment by ShivangiReja** [comment]
@timotheeguerin is correct. V3 generator depends on[ Azure.Core](https://www.nuget.org/packages/Azure.Core/) and we do not have any plan to support non Azure template.

**Comment by NelsonLamprecht** [comment]
> @timotheeguerin is correct. V3 generator depends on[ Azure.Core](https://www.nuget.org/packages/Azure.Core/) and we do not have any plan to support non Azure template.

Confirming that this package is now only for consuming azure rest services and not any rest services?
If so, I think it would be helpful for users to remove the section in the readme about "Use outside of the azure-sdk-net repo" and boldly proclaim autorest.csharp is not to be used to generate clients for non azure resources which appears to be a deviation from the proclaimed goals on the base autorest page: "The AutoRest tool generates client libraries for accessing RESTful web services". 

Maybe this package should be renamed autorest.csharp-azure since they are comingled?

**Comment by lmazuel** [comment]
Hi @NelsonLamprecht 

If you put aside the name of the runtime library, that indeed is called `Azure.Core`, what are the generated things you see that seems Azure specific to you? I want to separate the name of the runtime, with what actual code that is generated that you don't want to see.

@aanufriyev on this:
> We are actively using AutoRest to generate internal clients and if it is confirmed that non-Azure template will be dropped we will have to plan switch now, instead of using --v2 flag for some time.

Same question for you, what do you see in the generated code that feels Azure specific to you (runtime naming put aside)

**Comment by ShivangiReja** [comment]
@aanufriyev and @NelsonLamprecht:

>V3 generator depends on Azure.Core and we do not have any plan to support non Azure template.

I meant we depend on Azure.Core and we don't have any plan to support `Microsoft.Rest.ClientRuntime`(non Azure) library.

**Comment by borota** [comment]
@ShivangiReja @lmazuel Could you please create a sample *console* application running under Windows 10 (or Windows 2016) which uses a Rest client generated with V3? I am not able to.

**Comment by borota** [comment]
```
public ProductsRestClient(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri endpoint = null)
        {
            endpoint ??= new Uri("https://sbmsapiqa.gotostrata.com/SbmsWebApi/api");

            this.endpoint = endpoint;
            _clientDiagnostics = clientDiagnostics;
            _pipeline = pipeline;
        }
```
My guess is that a constructor like this can only be instantiated in an environment that either simulates Azure or it's some sort of  Azure 'service'. My exposure to Azure is extremely limited so I may not have the right lingo. From a *console* app, where do I get `clientDiagnostics` or `pipeline` from? My guess is, this is something that can only be provided if run inside of an Azure function/worker/service, something like that. See what I am getting at? This client can't be used outside of an Azure environment, can it?

**Comment by ShivangiReja** [comment]
@borota [Here ](https://github.com/ShivangiReja/AutorestCSharpConsoleApp)I created a console application using autorest c# V3 generator. Please take a look.

**Comment by borota** [comment]
How interesting @ShivangiReja, thank you. I really like the way the generated code looks now in V3.

**Comment by TheBoss3433** [comment]
> @borota [Here ](https://github.com/ShivangiReja/AutorestCSharpConsoleApp)I created a console application using autorest c# V3 generator. Please take a look.

Where is the console? I just see a "console" being initialized but that does not tell us where the parameters come from...

**Comment by ShivangiReja** [comment]
V3 generator also generates .csproj check [here](https://github.com/ShivangiReja/AutorestCSharpConsoleApp/blob/master/AutorestCSharpConsoleApp.csproj), it references  `Azure.Core` and `Microsoft.Azure.AutoRest.CSharp`. That's where these parameters come from.

**Comment by TheBoss3433** [comment]
> V3 generator also generates .csproj check [here](https://github.com/ShivangiReja/AutorestCSharpConsoleApp/blob/master/AutorestCSharpConsoleApp.csproj), it references  `Azure.Core` and `Microsoft.Azure.AutoRest.CSharp`. That's where these parameters come from.

Thanks for your reply, I did more digging and the issue is with blazon

**Comment by Saegusae** [comment]
Hey, just checking in one year later, I am having the exact same issues using this tool looks impossible to me outside an Azure environment. The documentation only shows azure-arm flavor of client initialization and nothing on how to provide those constructor parameters, namely `ClientDiagnostics` and `HttpPipeline`

Would appreciate input on this

**Comment by simeyla** [comment]
I tried to generate clients for Fedex API yesterday (obviously not an Azure service).
The Typescript client generated interfaces but the csharp version just made a useless stub for me with no entities.

If anyone is interested in looking into it, the API is at https://developer.fedex.com/api/en-us/catalog/rate/v1/docs.html 

The schema download link is top right of the page.

I used: 

    autorest --input-file=.\rate.json --csharp

I'm going back to NSwag, just wanted to try something new.
